nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-avro-record-utils/src/main/java/org/apache/nifi/avro/AvroTypeUtil.java

```diff
@@ -609,6 +623,9 @@ private static Object convertToAvroObject(final Object rawValue, final Schema fi
                 if (rawValue instanceof byte[]) {
                     return ByteBuffer.wrap((byte[]) rawValue);
                 }
+                if (rawValue instanceof String) {
+                    return ByteBuffer.wrap(((String) rawValue).getBytes(charset));
```

Comment:
I would prefer to avoid ByteBuffer here and instead use just byte[]

Reply:
In the clause above, a byte[] is wrapped in a ByteBuffer as well (that's where I got the code from), won't we be returning two different objects in that case?

Reply:
Whoops - my bad on this one. This is #convertToAvroObject, and I was thinking of #normalizeValue. In this case, we are converting into the object that Avro wants, so a ByteBuffer is the correct thing to do.